src/SourceBuild/content/test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs

```diff
@@ -0,0 +1,386 @@
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+using Xunit;
+using Xunit.Abstractions;
+using static System.Runtime.InteropServices.JavaScript.JSType;
+
+namespace Microsoft.DotNet.Installer.Tests;
+
+public class LinuxInstallerTests : IDisposable
+{
+    private readonly DockerHelper _dockerHelper;
+    private readonly string _tmpDir;
+    private readonly string _contextDir;
+
+    private readonly string[] RpmDistroImages =
+    [
+        "mcr.microsoft.com/dotnet-buildtools/prereqs:fedora-40"
+    ];
+
+    private readonly string[] DebDistroImages =
+    [
+        "mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-24.04"
+    ];
+
+    private const string NetStandard21RpmPackage = @"https://dotnetcli.blob.core.windows.net/dotnet/Runtime/3.1.0/netstandard-targeting-pack-2.1.0-x64.rpm";
+    private const string NetStandard21DebPackage = @"https://dotnetcli.blob.core.windows.net/dotnet/Runtime/3.1.0/netstandard-targeting-pack-2.1.0-x64.deb";
+
+    private enum DistroType
+    {
+        Rpm,
+        Deb
+    }
+
+    private ITestOutputHelper OutputHelper { get; set; }
+
+    public LinuxInstallerTests(ITestOutputHelper outputHelper)
+    {
+        OutputHelper = outputHelper;
+        _dockerHelper = new DockerHelper(OutputHelper);
+
+        _tmpDir = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+        Directory.CreateDirectory(_tmpDir);
+        _contextDir = Path.Combine(_tmpDir, Path.GetRandomFileName());
+        Directory.CreateDirectory(_contextDir);
+
+        InitializeContext();
+    }
+
+    public void Dispose()
+    {
+        try
+        {
+            Directory.Delete(_tmpDir, recursive: true);
+        }
+        catch
+        {
+        }
+    }
+
+    [Fact]
+    public void RunScenarioTestsForAllDistros()
+    {
+        if (Config.TestRpmPackages)
+        {
+            TestAllDistros(DistroType.Rpm);
+        }
+
+
+        if (Config.TestDebPackages)
+        {
+            TestAllDistros(DistroType.Deb);
+        }
+    }
+
+    private void TestAllDistros(DistroType distroType)
+    {
+        foreach (string image in (distroType == DistroType.Rpm ? RpmDistroImages : DebDistroImages))
+        {
+            try
+            {
+                OutputHelper.WriteLine($"Begin testing distro: {image}");
+                DistroTest(image, distroType);
+                OutputHelper.WriteLine($"Finished testing distro: {image}");
+            }
+            catch (Exception ex)
+            {
+                Assert.Fail($"Test failed for {image}: {ex}");
+            }
+        }
+    }
+
+    private void InitializeContext()
+    {
+        // For rpm enumerate RPM packages, excluding those that contain ".cm." in the name
+        List<string> rpmPackages = Directory.GetFiles(Config.AssetsDirectory, "*.rpm", SearchOption.AllDirectories)
+            .Where(p => !Path.GetFileName(p).Contains("-cm.") && !Path.GetFileName(p).EndsWith("azl.rpm"))
```

Comment:
Does this work if we were to test the Azure Linux 3.0 packages?

Reply:
Regular rpm packages have the exact same content as those with `azl.rpm` extension. Yes, it will work fine.

Reply:
`rpm` seems to only care if digests are ok. Will double-check this process with packages signed with regular Linux signatures.

Reply:
Yeah, `rpm` shows a warning when cert is not available, but happily installs the package. So, I think we can keep the existing code.